catalog/context/context.py

```diff
@@ -0,0 +1,55 @@
+import abc
+import collections
+import typing
+
+from django.db.models import QuerySet
+
+
+class Context(abc.ABC):
+
+    @abc.abstractmethod
+    def context(self) -> typing.Dict[str, typing.Any]:
+        ...
+
+
+class ModelContext(abc.ABC):
+
+    def __init__(self, qs: QuerySet):
+        self._qs = qs
+
+    def qs(self):
+        return self._qs
+
+    @abc.abstractmethod
+    def context(self) -> typing.Dict[str, typing.Any]:
+        ...
+
+
+Context.register(ModelContext)
+
+
+class Contexts(Context):
+
+    def __init__(self, *contexts: typing.List[Context]):
+        self.contexts = contexts
+
+    def context(self):
+        return collections.ChainMap(
+            *[ctx.context() for ctx in self.contexts]
```

Comment:
```suggestion
            iter(ctx.context() for ctx in self.contexts)
```

Reply:
I don't see the point. How `iter` can help here? `ChainMap` takes variadic dict number as arguments, so i have to unpack contexts list